### ISSUE TITLE
Massive framerate drops in a couple specific spots

### ISSUE THREAD

**Issue by stevethepocket** [ValveSoftware/portal2#128]
In sp_a1_wakeup, any time I'm inside GLaDOS's chamber or facing it from the hallway outside, the framerate drops to single digits. I don't know if it's just the sheer amount of clutter in the chamber, or a lack of optimization, or what.

Also, in sp_a3_crazy_box, the framerate slows to a crawl whenever I'm looking at the glass box. It might be something with the shader the glass material uses. Haven't noticed any issues with other refractive glass textures though, so I'm not sure how that would be the case.

System info:

```
Processor Information:
    Vendor:  GenuineIntel
    CPU Family:  0x6
    CPU Model:  0xf
    CPU Stepping:  0xb
    CPU Type:  0x0
    Speed:  2400 Mhz
    4 logical processors
    4 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Unsupported
    SSE42:  Unsupported

Network Information:
    Network Speed:  

Operating System Version:
    Ubuntu Trusty Tahr (development branch) (64 bit)
    Kernel Name:  Linux
    Kernel Version:  3.13.0-13-generic
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11500000
    X Window Manager:  KWin
    Steam Runtime Version:  steam-runtime-release_2014-02-05

Video Card:
    Driver:  X.Org Gallium 0.4 on AMD RV770

    Driver Version:  3.0 Mesa 10.1.0-rc1
    OpenGL Version: 3.0
    Desktop Color Depth: 24 bits per pixel
    Monitor Refresh Rate: 60 Hz
    VendorID:  0x1002
    DeviceID:  0x944a
    Number of Monitors:  1
    Number of Logical Video Cards:  1
    Primary Display Resolution:  1920 x 1080
    Desktop Resolution: 1920 x 1080
    Primary Display Size: 20.91" x 11.77"  (23.98" diag)
                                            53.1cm x 29.9cm  (60.9cm diag)
    Primary VRAM Not Detected

Sound card:
    Audio device: Realtek ALC1200

Memory:
    RAM:  3952 Mb

Miscellaneous:
    UI Language:  English
    LANG:  en_US.UTF-8
    Microphone:  Not set
    Total Hard Disk Space Available:  480490 Mb
    Largest Free Hard Disk Block:  348578 Mb
```

**Comment by MichaelAquilina** [comment]
I also have the same Issue. I get the same problem during the "escape" sequence with Wheatley too.

Kubuntu 13.10 64 bit
GeForce GT555m (running under primus)
Nvidia 331 Drivers

**Comment by kolonoskopia** [comment]
in my case changing antialias settings helped. no matter if decreasing or increasing, framerate is ok after that.

**Comment by ItsMeOnly** [comment]
It was the same for me, first impressions were discouraging because difference in performance between Windows (in XP) and Linux were significant (the latter behaved like on my trustworthy GF 210: under GLaDOS powerup the framerate would drop as low as 8 fps), but suddenly started working just fine, even with everything cranked up to maximum.

**Comment by mariodonofrio** [comment]
I'd be interested in hearing what difference turning vsync on/off makes, and reporting back to this thread.. use cl_showfps 2, go to map sp_a1_wakeup, after you initially enter the chamber (where framerate is at its worst) sit there and mat_vsync 0 and then 1 (or turn on/off from the video menu) and make a note of ms per frame and fps.
Thanks.

**Comment by ItsMeOnly** [comment]
For me it makes no distinguishable difference: it's oscillating around 30 fps with vsync on, goes to 34 fps without vsync. However on Shroedinger's catch level it's much worse: 11 fps (lowest observed: 8) when broken glass starts falling on the ground with or without vsync. Compare it with stable 17 fps/25 fps under Windows with the same settings. Under linux, the performance improves only without antialiasing and texture filtering (gradually improving to 25 fps).

Graphics: 440 GT /1 GB (nVidia drivers 331.38)
CPU: Intel Core i5 650 3.3GHz
RAM: 4 GB

**Comment by stevethepocket** [comment]
Yeah, turning vsync on and off didn't have any visible effect for me either. Neither, for that matter, does disabling anisotropic filtering or antialiasing. Which is especially odd since turning off antialiasing nearly doubles the framerate in areas with much less detail _to_ antialias.
